boot-mongodb-elasticsearch/src/main/java/com/example/mongoes/web/service/RestaurantService.java

```diff
@@ -40,16 +40,25 @@
 import reactor.core.publisher.Mono;
 
 @Service
-@Slf4j
-@RequiredArgsConstructor
 public class RestaurantService {
 
+    private static final Logger log = LoggerFactory.getLogger(RestaurantService.class);
     private final RestaurantRepository restaurantRepository;
     private final RestaurantESRepository restaurantESRepository;
     private final ChangeStreamResumeRepository changeStreamResumeRepository;
-
     private final ReactiveMongoTemplate reactiveMongoTemplate;
 
+    public RestaurantService(
+            RestaurantRepository restaurantRepository,
+            RestaurantESRepository restaurantESRepository,
+            ChangeStreamResumeRepository changeStreamResumeRepository,
+            ReactiveMongoTemplate reactiveMongoTemplate) {
+        this.restaurantRepository = restaurantRepository;
+        this.restaurantESRepository = restaurantESRepository;
+        this.changeStreamResumeRepository = changeStreamResumeRepository;
+        this.reactiveMongoTemplate = reactiveMongoTemplate;
+    }
+
```

Comment:
@coderabbitai, can you please rewrite `private Flux<Restaurant> saveAll(List<String> restaurantStringList)` method effectively and which is performant